### DISCUSSION TITLE
5.0.0

### DISCUSSION THREAD

**Discussion by pubpub-zz** [py-pdf/pypdf#2854] [Announcements]
## Version 5.0.0, 2024-09-15

This version drops support for Python 3.7 (not maintained since July 2023), PdfMerger (use PdfWriter instead) and AnnotationBuilder (use annotations instead).

### Deprecations (DEP)
- Remove the deprecated PfdMerger and AnnotationBuilder classes and other deprecations cleanup (https://github.com/py-pdf/pypdf/pull/2813)
- Drop Python 3.7 support (https://github.com/py-pdf/pypdf/pull/2793)

### New Features (ENH)
- Add capability to remove /Info from PDF (https://github.com/py-pdf/pypdf/pull/2820)
- Add incremental capability to PdfWriter (https://github.com/py-pdf/pypdf/pull/2811)
- Add UniGB-UTF16 encodings (https://github.com/py-pdf/pypdf/pull/2819)
- Accept utf strings for metadata (https://github.com/py-pdf/pypdf/pull/2802)
- Report PdfReadError instead of RecursionError (https://github.com/py-pdf/pypdf/pull/2800)
- Compress PDF files merging identical objects (https://github.com/py-pdf/pypdf/pull/2795)

### Bug Fixes (BUG)
- Fix sheared image (https://github.com/py-pdf/pypdf/pull/2801)

### Robustness (ROB)
- Robustify .set_data() (https://github.com/py-pdf/pypdf/pull/2821)
- Raise PdfReadError when missing /Root in trailer (https://github.com/py-pdf/pypdf/pull/2808)
- Fix extract_text() issues on damaged PDFs (https://github.com/py-pdf/pypdf/pull/2760)
- Handle images with empty data when processing an image from bytes (https://github.com/py-pdf/pypdf/pull/2786)

### Developer Experience (DEV)
- Fix coverage uploads (https://github.com/py-pdf/pypdf/pull/2832)
- Test against Python 3.13 (https://github.com/py-pdf/pypdf/pull/2776)

[Full Changelog](https://github.com/py-pdf/pypdf/compare/4.3.1...5.0.0)

<hr /><em>This discussion was created from the release <a href='https://github.com/py-pdf/pypdf/releases/tag/'>5.0.0</a>.</em>

**Comment by Debajyati** [comment]
The most amazing thing is the divine intervention. I remember I was trying out PYPDF2 the last year in a python project and today suddenly I have got an idea to build a CLI upon it. I was looking at the docs and realised there's a version 3, okay there is even a version 4 also!
Oh man! I came to the repo and saw version 5.0.0 just released a few moments ago. For all the maintainers here, it is great to see efforts to keep an open-source project alive! I am saying it so because some people told me that PYPDF is going to get deprecated soon. They all are wrong! Yeah!

**Reply by stefan6419846** [reply]
Thanks for the kind words. Being a large user of the library myself, I am glad that Martin and *pubpub-zz* started fixing lots of its bugs. I still remember the times where I basically had PyPDF2, PyPDF3 and PyPDF4 installed in the same environment as PyPDF2 was not maintained and no clear successor was visible.
